src/pam_google_authenticator.c

```diff
@@ -1707,6 +1720,27 @@ static int google_authenticator(pam_handle_t *pamh, int flags,
         break;
       }
       if (!pw) {
+        char* host;
```

Comment:
This block of code should be in a helper function that returns a bool of whether the request can be revalidated.  That function should be called from within the switch statement above for the case 1 condition:

```
      case 1: // Extract possible scratch code
        if (params.pass_mode == USE_FIRST_PASS ||
            params.pass_mode == TRY_FIRST_PASS) {
          pw = get_first_pass(pamh);
        }
        if (!pw && check_revalidated(pamh)) {
          goto success;
        }
        break;
```